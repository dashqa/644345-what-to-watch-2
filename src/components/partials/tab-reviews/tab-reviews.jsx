import React from "react";
import PropTypes from "prop-types";

import {formatDate} from "@utils";
import withDividedComments from "@hocs/with-divided-comments/with-divided-comments";


const TabReviews = ({dividedComments}) => {
  return (
    <div className="movie-card__reviews movie-card__row">
      {Object.entries(dividedComments).map(([key, comments]) => (
        <div
          className="movie-card__reviews-col"
          key={key}
        >
          {comments.map(({id, comment, user, date, rating}) => {
            return (
              <div
                className="review"
                key={id}
              >
                <blockquote className="review__quote">
                  <p className="review__text">{comment}</p>

                  <footer className="review__details">
                    <cite className="review__author">{user.name}</cite>
                    <time className="review__date" dateTime={formatDate(date, `YYYY-MM-DD`)}>
                      {formatDate(date, `MMMM Do, YYYY`)}</time>
                  </footer>
                </blockquote>

                <div className="review__rating">{rating}</div>
              </div>
            );
          })}
        </div>
      ))}
    </div>
  );
};

TabReviews.deafultProps = {
  dividedComments: {},
};

TabReviews.propTypes = {
  dividedComments: PropTypes.objectOf(
      PropTypes.arrayOf(
          PropTypes.object
      ))
};

export {TabReviews};
export default withDividedComments(TabReviews);
