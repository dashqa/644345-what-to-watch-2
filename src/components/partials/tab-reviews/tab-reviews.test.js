import React from 'react';
import renderer from 'react-test-renderer';
import {TabReviews} from './tab-reviews';
import moment from "moment";

const props = {
  dividedComments: {
    odd: [{
      id: 1,
      user: {
        id: 1,
        name: `Dasha`
      },
      comment: `Hello!!`,
      date: new Date(2019, 11, 14),
      rating: 3.5
    }],
    even: [{
      id: 2,
      user: {
        id: 4,
        name: `Masha`
      },
      comment: `Bye`,
      date: new Date(2019, 11, 27),
      rating: 1.1
    }]
  },
};

it(`Reviews tab correctly renders after relaunch`, () => {
  const tree = renderer.create(<TabReviews {...props}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
