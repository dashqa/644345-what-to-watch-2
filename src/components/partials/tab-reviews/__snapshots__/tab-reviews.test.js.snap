// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reviews tab correctly renders after relaunch 1`] = `
<div
  className="movie-card__reviews movie-card__row"
>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Hello!!
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Dasha
          </cite>
          <time
            className="review__date"
            dateTime="2019-11-24"
          >
            November 24th, 2019
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        3.5
      </div>
    </div>
  </div>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Bye
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Masha
          </cite>
          <time
            className="review__date"
            dateTime="2019-11-09"
          >
            November 9th, 2019
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        1.1
      </div>
    </div>
  </div>
</div>
`;
